lms/djangoapps/grades/signals/handlers.py

```diff
@@ -120,16 +120,18 @@ def disconnect_submissions_signal_receiver(signal):
     """
     if signal == score_set:
         handler = submissions_score_set_handler
+        dispatch_uid = 'submissions_score_set_handler'
     else:
         if signal != score_reset:
             raise ValueError("This context manager only handles score_set and score_reset signals.")
         handler = submissions_score_reset_handler
+        dispatch_uid = 'submissions_score_reset_handler'
 
-    signal.disconnect(handler)
```

Comment:
Disconnecting by function reference wasn't working because of the `receiver` decorator, fixed to disconnect by ID instead.  Needed to resolve issues in `lms/djangoapps/grades/tests/test_signals.py`.